Marlin/Marlin_main.cpp

```diff
@@ -6005,7 +6005,11 @@ inline void gcode_M428() {
   }
 
   if (!err) {
-    sync_plan_position();
+    #if ENABLED(DELTA) && ENABLED(SCARA)
```

Comment:
@thinkyhead 
I'm sorry, I tried to understand your instruction and content of this merged PR,
but I stumbled on the first step.
Why isn't this `ENABLED(DELTA) || ENABLED(SCARA)` but `ENABLED(DELTA) && ENABLED(SCARA)`?
[This combination is forbidden in SanityCheck](https://github.com/MarlinFirmware/Marlin/blob/RCBugFix/Marlin/SanityCheck.h#L402).